### PR TITLE
Issue #11277: update code base to have javadoc tag to explain noinspection content

### LINKED ISSUE #11277: update code base to have javadoc tag to explain noinspection tag content
detected at https://github.com/checkstyle/checkstyle/pull/11206#discussion_r797996787

code:
https://github.com/checkstyle/checkstyle/blob/d4228cc1c04f6e18392948986b605cf87738c951/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L628-L635

has problems that such technical excuse of to suppress IDEA inspection is rendered in final javadoc and can confuse users.
we try to keep reasons of suppressions but in same time we do not want to expose it to users.

proposed solution is too use custom tag for this:
```
$ git diff
diff --git a/pom.xml b/pom.xml
index 31c2853..65fddda 100644
--- a/pom.xml
+++ b/pom.xml
@@ -433,6 +433,10 @@
                 <name>noinspection</name>
                 <placement>X</placement>
               </tag>
+              <tag>
+                <name>noinspectionreason</name>
+                <placement>X</placement>
+              </tag>
             </tags>
           </configuration>
         </plugin>
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/Main.java b/src/main/java/com/puppycrawl/tools/checkstyle/Main.java
index fec6cbd..f216a44 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/Main.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/Main.java
@@ -627,8 +627,10 @@ public final class Main {
 
         /**
          * The checker threads number.
-         * Suppression: CanBeFinal - we use picocli and it use  reflection to manage such fields
-         * This option has been skipped for CLI options intentionally.
+         *
+         * @noinspectionreason CanBeFinal - we use picocli and it use
+         *     reflection to manage such fields
+         *     This option has been skipped for CLI options intentionally.
          *
          * @noinspection CanBeFinal
          */
diff --git a/config/intellij-idea-inspections.xml b/config/intellij-idea-inspections.xml
index de97b29..35a1ab0 100644
--- a/config/intellij-idea-inspections.xml
+++ b/config/intellij-idea-inspections.xml
@@ -2797,7 +2797,7 @@
     <option name="IGNORE_JAVADOC_PERIOD" value="true"/>
     <option name="IGNORE_DUPLICATED_THROWS" value="false"/>
     <option name="IGNORE_POINT_TO_ITSELF" value="false"/>
-    <option name="myAdditionalJavadocTags" value=""/>
+    <option name="myAdditionalJavadocTags" value="noinspectionreason"/>
   </inspection_tool>
   <inspection_tool class="JavaFxColorRgb" enabled="false" level="WARNING"
                    enabled_by_default="false"/>

```

### PR DESCRIPTION
**Pull request by nick-mancuso**
Includes: ProhibitedExceptionThrown, ChainOfInstanceofChecks, FieldNotUsedInToString, SuspiciousArrayCast, ClassWithTooManyDependencies. Part of #11277 .

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java [hunk 1]
```diff
@@ -76,6 +76,7 @@
  * TreeWalkerTest.
  *
  * @noinspection ClassWithTooManyDependencies because we are less strict with tests.
+ * @noinspectionreason complex tests require a large number of imports
```
[the remark this answers is not in the document]
Comment (nick-mancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java:79]: Mistake while splitting the original PR up, hence the self assignment

[the remark this answers is not in the document]
Comment (nick-mancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java]: Fixed.

**Review by pbludov — APPROVED** [review]

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java [hunk 2]
```diff
@@ -777,6 +779,8 @@ public void testCatchErrorInProcessFilesMethod() throws Exception {
              * Test is checking catch clause when exception is thrown.
              *
              * @noinspection ProhibitedExceptionThrown
+             * @noinspectionreason ProhibitedExceptionThrown - There is no other way to obey
+             *      haltOnException field.
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java:783]: This and the others do not appear to be testing haltOnException...

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java (hunk 2 above)
Comment (nick-mancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java]: Yes, you are right. Done.

**Review by rnveach — APPROVED** [review]

**Comment by rnveach** [comment]
@nick-mancuso There is a conflict.

**Comment by romani** [comment]
conflict is resolved

**Pull request merged**